File: stald/stald.py
import sys

lines = [l.strip() for l in sys.stdin.readlines()]
c = 0
counts = [0]


for line in lines:
    if line == 'Får ind':
        c += 1
    elif line == 'Får ud':
        c -= 1
    counts.append(c)

print(max(counts) - min(counts))

File: stald/stald.rs
use std::io::{self, Read};

fn main() -> io::Result<()> {
    let mut buffer = String::new();
    let mut stdin = io::stdin(); // We get `Stdin` here.
    stdin.read_to_string(&mut buffer)?;
    let lines = buffer.lines();

    let mut c = 0;
    let mut counts = vec![0];
    for line in lines {
        if line == "Får ind" {
            c += 1;
        } else {
            c -= 1;
        }
        counts.push(c);
    }
    println!("{}", counts.iter().max().unwrap() - counts.iter().min().unwrap());

    Ok(())
}
